### PR TITLE
Model parallelism: Moving labels to same devices as the logits are

### PR DESCRIPTION
**Pull request by shahad-mahmud**
As suggested in the https://github.com/huggingface/transformers/issues/22561, moving the labels to the same device as the logits are for the Data2Vec Text, ESM, Longformer and LongT5 models.

@sgugger Can you please review?

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks a lot for your contribution!
